lib/commons/dom/get-root-node.js

```diff
@@ -7,12 +7,6 @@
  * @instance
  * @param {Element} node
  * @returns {DocumentFragment|Document}
+ * @deprecated use axe.utils.getRootNode
  */
-dom.getRootNode = function(node) {
-	var doc = (node.getRootNode && node.getRootNode()) || document; // this is for backwards compatibility
-	if (doc === node) {
-		// disconnected node
-		doc = document;
-	}
-	return doc;
-};
+dom.getRootNode = axe.utils.getRootNode;
```

Comment:
Should we add a warning here when this method is called? Just aliasing it doesn't seem to give users any reason _not_ to use it.

Something like:

```js
let didWarn = false
dom.getRootNode = (...args) => {
  if (!didWarn) {
    console.warn('axe-core: dom.getRootNode has been deprecated; use utils.getRootNode instead.')
    didWarn = true
  }
  axe.utils.getRootNode(...args)
}
```

Reply:
I like this.

Reason I did not do this is right away is, we are bypassing the function to `axe.utils` for the time being. 

The `jsdoc` `@deprecated` notice is a flag for the dev's (us) to take some action on a later date.

There are multiple places with in the code base still referring to `axe.commons.dom.getRootNode`, until we change all these references internally first,  to `axe.utils.getRootNode` believe we should not warn on the console.

And before all the internal changes come be made, this PR needs to be merged.

Also reckon a deprecation notice should dictate when an API is going to be deprecated, so we need to agree on that, something like `this function has been deprecated from v3.2.0, use newFunction`.

Happy to add this, once we reach that consensus.

@marcysutton @WilcoFiers - thoughts?

Thanks @stephenmathieson